### PR TITLE
Properly enter loop if starting point is 0, i.E undefined.

### LINKED ISSUE #843: AB-Loop does not actually loop 2 : Electric Loopaloo.
**Describe the bug**
Any song played on channel 0 does not loop trough the ABLoop system.

**To reproduce**
1. Have an AB-Loop configured song.
2. Let the song play till the looppoint is supposed to hit
3. Notice how it does not loop

**Expected behavior**
AB-Looping is applied correctly.

**Sample file**
[track.zip](https://github.com/AttorneyOnline/AO2-Client/files/9254534/track.2.zip)

**Version:**
 - OS: Windows 10 Pro 21H2
 - Version or branch: master

**Additional context**
Fml.

### PR DESCRIPTION
**Pull request by Salanto**
closes #843

### CODE REVIEW

File: src/aomusicplayer.cpp [hunk 1]
```diff
@@ -219,7 +219,7 @@ void AOMusicPlayer::set_looping(bool loop_song, int channel)
     loop_sync[channel] = 0;
   }
 
-  if (loop_start[channel] > 0) {
+  if (loop_start[channel] >= 0) {
     if (loop_end[channel] > 0 && (loop_end[channel] > loop_start[channel]))
```
Comment (TrickyLeifa) [inline src/aomusicplayer.cpp:223]: You most likely should do the same check for loop_end.

File: src/aomusicplayer.cpp [hunk 2]
```diff
@@ -219,20 +219,20 @@ void AOMusicPlayer::set_looping(bool loop_song, int channel)
...
+          loop_end[channel], loopProc, &loop_start[channel]);
     }
     else
     {
       //Loop when the end of the file is reached.
       loop_sync[channel] = BASS_ChannelSetSync(
-          m_stream_list[channel], BASS_SYNC_POS | BASS_SYNC_MIXTIME,
-          0 , loopProc, &loop_start[channel]);
+          m_stream_list[channel], BASS_SYNC_END | BASS_SYNC_MIXTIME,
+          0, loopProc, &loop_start[channel]);
```
Comment (oldmud0) [inline src/aomusicplayer.cpp:235]: if you want to invert this if statement make sure you also move the comments

**Review by oldmud0 — COMMENTED** [review]
i guess this makes sense yeah

**Comment by Crystalwarrior** [comment]
@stonedDiscord any idea why the android build fails?

**Review by Crystalwarrior — APPROVED** [review]

File: src/aomusicplayer.cpp (hunk 2 above)
Comment (TrickyLeifa) [reply src/aomusicplayer.cpp:235]: The comments were on the right spots. The function call however, was inverted.

**Comment by Salanto** [comment]
@Crystalwarrior It's clearly a CI issue, so open an issue and don't put a CI fix in here.

File: src/aomusicplayer.cpp (hunk 2 above)
Comment (Salanto) [reply src/aomusicplayer.cpp:235]: Sorry!

**Pull request merged**